packages/iris-grid/src/IrisGridProxyModel.ts

```diff
@@ -476,6 +493,74 @@ class IrisGridProxyModel extends IrisGridModel {
     this.model.filter = filter;
   }
 
+  get partition(): readonly unknown[] {
+    return this.originalModel.partition;
+  }
+
+  set partition(partition: readonly unknown[]) {
+    log.debug('set partition', partition);
```

Comment:
You should add a check in here to see if our original model is a partitioned table model, and if not just throw. E.g:
```suggestion
    if (!isIrisGridPartitionedTableModel(this.originalModel)) {
      throw new Error('Partitions are not available');
    }
    log.debug('set partition', partition);
```

Reply:
I also need to be able to set the partition for `IrisGridTableModel` for the partitioned parquet tables

Reply:
Ahh we should have a flag for checking if the model is partitionable then. `isPartitionRequired` I think is the right flag to check? I don't think there's a case where it's partitionable but a partition is not required is there?

Reply:
The original `isPartitionRequired` is actually equal to `table.isUncoalesced` (for parquet tables), which returns `false` after a partition is set for the first time. Is there some other way I can check whether a partition is required that I can use to replace the existing implementation of `isPartitionRequired`?